src/Fixer/Whitespace/BlankLineBeforeStatementFixer.php

```diff
@@ -328,4 +311,50 @@ protected function createConfigurationDefinition()
                 ->getOption(),
         ]);
     }
+
+    /**
+     * @param int $prevNonWhitespace
+     *
+     * @return bool
+     */
+    private function shouldAddBlankLine(Tokens $tokens, $prevNonWhitespace)
+    {
+        $prevNonWhitespaceToken = $tokens[$prevNonWhitespace];
+
+        if ($prevNonWhitespaceToken->isComment()) {
+            for ($j = $prevNonWhitespace - 1; $j >= 0; --$j) {
```

Comment:
Do we really need this loop? There will be max 2 iterations, because of the 2nd `if`, right? Either the token is not whitespace, so we return there or token is whitespace which means next one cannot be.

Reply:
I'll add a test, but I hope this code will also fix;

```php
yield $a; /* a *//* b */     /* c */       /* d *//* e *//* etc */
yield $b;
```

to 

```php
yield $a; /* a *//* b */     /* c */       /* d *//* e *//* etc */

yield $b;
```

Reply:
it was bugged indeed! good catch, please see the follow up commit :)

Reply:
Oh, no wI see the point of having the loop - it can be crazy series of whitespace and comments 👍